src/utils.rs

```diff
@@ -268,15 +268,37 @@ pub(crate) fn now_systime(clock: &dyn Clock) -> SystemTime {
         .expect("Failed to convert Clock time to SystemTime")
 }
 
+/// Computes the "index key" (lowest bound) for an SST index block, ie a key that's greater
+/// than all keys in the previous block and less than or equal to all keys in the new block
+pub(crate) fn index_key(prev_block_last_key: Option<Bytes>, this_block_first_key: &Bytes) -> Bytes {
+    if prev_block_last_key.is_none() {
+        Bytes::new()
```

Comment:
pretty sure this is right but just as a sanity check: it's ok to use empty bytes for the index key of the first block since the index keys are only used to determine the ranges of blocks within an SST, and it tracks the actual first key of the overall SST separately